src/runtime/HalideRuntime.h

```diff
@@ -541,6 +546,38 @@ extern int halide_copy_to_host(void *user_context, struct halide_buffer_t *buf);
 extern int halide_copy_to_device(void *user_context, struct halide_buffer_t *buf,
                                  const struct halide_device_interface_t *device_interface);
 
+/** Copy data from one buffer to another. The buffers may have
+ * different shapes and sizes, but the destination buffer's shape must
+ * be contained within the source buffer's shape. The source data is
+ * pulled from either device or host memory on the source, depending
+ * on the dirty flags. host is preferred if both are valid. The
+ * dst_device_interface parameter controls the destination memory
+ * space. NULL means host memory. */
+extern int halide_buffer_copy(void *user_context, struct halide_buffer_t *src,
+                              const struct halide_device_interface_t *dst_device_interface,
+                              struct halide_buffer_t *dst);
+
+/** Give the destination buffer a device allocation which is an alias
+ * for the same coordinate range in the source buffer. Modifies the
+ * device, device_interface, and the device_dirty flag only. Only
+ * supported by some device APIs (others will return
+ * halide_error_code_device_crop_unsupported). Call
+ * halide_device_release_crop instead of halide_device_free to clean
+ * up resources associated with the cropped view. Do not call
+ * device_free on the source buffer while the destination buffer still
```

Comment:
Probably should say the parent cannot be freed by any means. (It could be a combined allocation or a wrapped native, etc.)